### PR TITLE
add srun for running e3sm_diags

### LINKED ISSUE #485: Unified rc12 and rc14 testing: `e3sm_diags` takes too long on Chrysalis
Even after setting `walltime = "4:00:00"` for the `[[ atm_monthly_180x360_aave ]]` subtask in `tests/integration/generated/test_complete_run_chrysalis.cfg`, the following jobs still run out of time:

```
e3sm_diags_atm_monthly_180x360_aave_model_vs_obs_1850-1851
e3sm_diags_atm_monthly_180x360_aave_model_vs_obs_1850-1853
e3sm_diags_atm_monthly_180x360_aave_model_vs_obs_1852-1853
```

This is a change from `walltime = "2:00:00"` in https://github.com/E3SM-Project/zppy/blob/main/tests/integration/generated/test_complete_run_chrysalis.cfg#L93

It also took significantly longer to get compute nodes with this extended time limit.

### PR DESCRIPTION
**Pull request by chengzhuzhang**
Fixes #485 for the PMI2_Init error and e3sm_diags slowness on Chrysalis.

### CODE REVIEW

**Comment by chengzhuzhang** [comment]
@forsyth2 while adding `srun -n 1` to the e3sm_diags bash script works through submitting the standalone bash via `sbatch`.  I hit error:
```
===== RUN E3SM DIAGS =====

slurmstepd: error: execve(): python: No such file or directory
srun: error: chr-0311: task 0: Exited with exit code 2
```
If submit the job via `zppy`, any insight to have this fixed?

**Comment by forsyth2** [comment]
@chengzhuzhang That's really quite perplexing. The only thing I can think of is that the correct environment isn't being picked up when running with `srun`. But it doesn't even recognize `python` as a command here. And we've used `srun` plenty of other places in `zppy` without a problem...

File: zppy/e3sm_diags.py [hunk 1]
```diff
@@ -222,7 +222,7 @@ def e3sm_diags(config, scriptDir, existing_bundles, job_ids_file):  # noqa: C901
                 p.pprint(c)
                 p.pprint(s)
 
-            export = "NONE"
+            export = "ALL"
```
Comment (chengzhuzhang) [inline zppy/e3sm_diags.py:225]: Set export to "ALL" to load all environment.

Comment (forsyth2) [reply zppy/e3sm_diags.py:225]: Ah, that makes sense.

**Comment by forsyth2** [comment]
@chengzhuzhang I was able to reproduce your initial error and was able to get E3SM Diags working with your `export` line fix. I'm currently running with these changes to double check the elapsed time returns to a reasonable duration.

**Comment by chengzhuzhang** [comment]
@forsyth2 I think this PR is ready to review. With export="ALL", it loads the correct environment and e3sm_diags run launched successfully. I'm testing zppy on `Chrysalis`. Could you help test this branch with unified rc14 on `perlmutter` and `Compy`, if things go okay, please go ahead and release a new zppy rc..thank you.

**Comment by chengzhuzhang** [comment]
> @chengzhuzhang I was able to reproduce your initial error and was able to get E3SM Diags working with your `export` line fix. I'm currently running with these changes to double check the elapsed time returns to a reasonable duration.

Sounds good. the model vs obs task elapsed time fell within one hour on `Chrysalis`. Hope there is no change for `perlmutter` and `compy`

**Comment by forsyth2** [comment]
> Could you help test this branch with unified rc14 on `perlmutter` and `Compy`, if things go okay, please go ahead and release a new zppy rc..thank you.

Yes, I'll run on those and then hopefully make the last RC before final release.

**Comment by forsyth2** [comment]
Confirmed that, on Chrysalis, times indeed have been reduced:
```
$ tail -n 3 e3sm_diags_atm_monthly_180x360_aave_*.o*
==> e3sm_diags_atm_monthly_180x360_aave_environment_commands_model_vs_obs_1850-1851.o385043 <==
==============================================
Elapsed time: 43 seconds
==============================================

==> e3sm_diags_atm_monthly_180x360_aave_environment_commands_model_vs_obs_1850-1853.o385045 <==
==============================================
Elapsed time: 43 seconds
==============================================

==> e3sm_diags_atm_monthly_180x360_aave_environment_commands_model_vs_obs_1852-1853.o385044 <==
==============================================
Elapsed time: 43 seconds
==============================================

==> e3sm_diags_atm_monthly_180x360_aave_model_vs_obs_1850-1851.o385040 <==
==============================================
Elapsed time: 3562 seconds
==============================================

==> e3sm_diags_atm_monthly_180x360_aave_model_vs_obs_1850-1853.o385042 <==
==============================================
Elapsed time: 3604 seconds
==============================================

==> e3sm_diags_atm_monthly_180x360_aave_model_vs_obs_1852-1853.o385041 <==
==============================================
Elapsed time: 3586 seconds
==============================================

==> e3sm_diags_atm_monthly_180x360_aave_mvm_model_vs_model_1852-1853_vs_1850-1851.o385047 <==
==============================================
Elapsed time: 682 seconds
==============================================

==> e3sm_diags_atm_monthly_180x360_aave_tc_analysis_model_vs_obs_1850-1851.o385046 <==
==============================================
Elapsed time: 40 seconds
==============================================
```

**Review by forsyth2 — APPROVED** [review]
The `complete_run` test passes on Perlmutter and Compy still, so I think this ready to merge.

**Pull request merged**